src/nl/hannahsten/texifyidea/refactoring/inlinefile/LatexInlineFileHandler.kt

```diff
@@ -0,0 +1,117 @@
+
+/**
+ * Allows for inlining an include command. Does not do the inlining itself.
+ *
+ * @author jojo2357
+ */
+class LatexInlineFileHandler : InlineActionHandler() {
+
+    override fun isEnabledForLanguage(language: Language?): Boolean {
+        return language == LatexLanguage
+    }
+
+    override fun canInlineElement(element: PsiElement?): Boolean {
+        return canInlineLatexElement(element)
+    }
+
+    override fun inlineElement(project: Project, editor: Editor?, element: PsiElement) {
+        // Resolve the file to be inlined
+        val inlineFile: LatexFile = resolveInlineFile(element) ?: return
+
+        val dialog = LatexInlineFileDialog(
+            project,
+            inlineFile,
+            getReference(element, editor),
+            editor != null
+        )
+
+        if (dialog.myOccurrencesNumber > 0) {
+            if (ApplicationManager.getApplication().isUnitTestMode) {
+                try {
+                    dialog.doAction()
+                }
+                finally {
+                    dialog.close(DialogWrapper.OK_EXIT_CODE, true)
+                }
+            }
+            else {
+                dialog.show()
+            }
+        }
+        else {
+            // TODO No ocurrences, what to do?
+        }
+    }
+}
+
+/**
+ * This is static only so that the unit tests can use it, and also since it can be static
+ */
+fun canInlineLatexElement(element: PsiElement?): Boolean {
+    val out = when (element) {
+        is LatexFile -> {
+            element.containingFile.isLatexFile()
+        }
+
+        is LatexCommandsImpl -> {
+            ((element as LatexCommandsImpl).getName() == "\\input" && (element as PsiElementBase).canNavigateToSource())
+        }
+
+        else -> true
+    }
+
+    return out
+}
+
+fun resolveInlineFile(element: PsiElement) = when (element) {
+    // If we tried to inline the file in file view, or the argument in `input`
+    is LatexFile -> element
+    // If we tried to inline the `input` command
+    is LatexCommandsImpl -> element.containingFile.parent?.findFile(
+        if ((element as LatexCommandsImpl).requiredParameters[0].matches(
+                ".*\\.\\w{3}$".toRegex()
+            )
+        ) (element as LatexCommandsImpl).requiredParameters[0]
+        else (element as LatexCommandsImpl).requiredParameters[0] + ".tex"
+    ) as? LatexFile
+
+    else -> null
+}
+
+/**
+ * Resolves the command that the inlining was called on. If called from file view then expected return is null
+ */
+fun getReference(
+    element: PsiElement,
+    editor: Editor?
+): PsiElement? {
+    return if (element is LatexCommandsImpl)
+        element
+    else if (editor != null) {
+        if (getElementAtCaret(editor)?.parent?.parent?.parent is LatexCommandsImpl)
+            getElementAtCaret(editor)?.parent?.parent?.parent
+        else if (getElementAtCaret(editor)?.parent?.parent?.parent?.parent?.parent is LatexCommandsImpl)
```

Comment:
Please use `firstParentOfType`, because the psi structure may change

Reply:
I honestly don't know how to get the parent command that the latexfile argument comes from, this works so I could really use a crash course on how to resolve these things (as per your other comment)

Reply:
I think `getElementAtCaret(editor)?.firstParentOfType(LatexCommands)` would work just fine, not in general but good enough here
(Please use `LatexCommands` instead of `LatexCommandsImpl`)

Reply:
Lol IJ kept autocasting to LatexCommandsImpl so thats why I chose it \shrug. I will look into it

Reply:
`val ref: LatexCommands? = getElementAtCaret(editor)?.firstParentOfType(LatexCommands)` says `Classifier 'LatexCommands' does not have a companion object, and thus must be initialized here`

Reply:
Nevermind, solved that